src/main/java/io/supertokens/session/accessToken/AccessToken.java

```diff
@@ -133,6 +140,26 @@ public static AccessTokenInfo getInfoFromAccessTokenWithoutVerifying(@Nonnull St
     }
 
     public static TokenInfo createNewAccessToken(@Nonnull Main main, @Nonnull String sessionHandle,
+            @Nonnull String userId, @Nonnull String refreshTokenHash1, @Nullable String parentRefreshTokenHash1,
+            @Nonnull JsonObject userData, @Nonnull JsonObject grantPayload, @Nullable String antiCsrfToken, long lmrt,
+            @Nullable Long expiryTime)
+            throws StorageQueryException, StorageTransactionLogicException, InvalidKeyException,
+            NoSuchAlgorithmException, UnsupportedEncodingException, InvalidKeySpecException, SignatureException {
+
+        Utils.PubPriKey signingKey = new Utils.PubPriKey(
+                AccessTokenSigningKey.getInstance(main).getLatestIssuedKey().value);
+        long now = System.currentTimeMillis();
+        if (expiryTime == null) {
+            expiryTime = now + Config.getConfig(main).getAccessTokenValidity();
+        }
+        AccessTokenInfo accessToken = new AccessTokenInfo(sessionHandle, userId, refreshTokenHash1, expiryTime,
+                parentRefreshTokenHash1, userData, grantPayload, antiCsrfToken, now, lmrt);
+        String token = JWT.createJWT(new Gson().toJsonTree(accessToken), signingKey.privateKey, VERSION.V3);
+        return new TokenInfo(token, expiryTime, now);
+
+    }
+
+    public static TokenInfo createNewAccessTokenV2(@Nonnull Main main, @Nonnull String sessionHandle,
```

Comment:
This should just call `createNewAccessToken` with passing `null` in `grantPayload` param?

Reply:
The created JWT has a different header and this way it can have the `grantPayload` param be `@Nonnull`.  Also, it was more consistent with the V2-V1 methods this way. I'm happy to update it if that's what you prefer.